onnxscript/rewriter/llama_rule_sets.py

```diff
@@ -26,9 +26,12 @@ def pattern(self, op, x):
     def rewrite(self, op, x: ir.Value):
         return op.Identity(x)
 
-    def check(self, context, x) -> bool:
+    def check(self, context, x) -> orp.MatchResult:
         del context  # Unused
-        return ir_utils.has_rank(x, 1)
+        check_result = orp.MatchResult()
+        if not ir_utils.has_rank(x, 1):
+            return check_result.fail("Input is not 1D")
+        return check_result
```

Comment:
Another nit thought: it might be slightly better if this looked like
```py
   if ...
      return orp.failure("reason")
   return orp.SUCCESS
```
Just a stylistic suggestion to keep in mind. I realize this pattern has already been used in many places down below and may not be worth changing at this point.